test/test_xla_backend.py

```diff
@@ -0,0 +1,442 @@
+    try:
+      filename = random_filename()
+      port = random.randrange(49152, 65535)
+      os.environ['XRT_WORKERS'] = f'localservice:0;grpc://localhost:{port}'
+      # `world_size` needs to be 1, so that the rendezvous would only
+      # wait for 1 participant, which is this program itself.
+      # `init_method` is needed to provide a internal storage to comfort
+      # torch.distributed.
+      dist.init_process_group(
+          'xla', rank=0, world_size=1, init_method='file:///tmp/' + filename)
+      succeeded = True
+      break
+    except RuntimeError as err:
+      os.remove('/tmp/' + filename)
+      if 'Could not start gRPC server' in str(err):
+        continue
+      raise
+  assert succeeded, f'Failed to init torch.distributed after {retries} retries'
+  print('init_torch_distributed done')
+
+
+@pytest.fixture(scope='module', autouse=True)
+def setup(request):
+  init_torch_distributed()
+
+
+def set_world_size(size):
+  world_pg = dist.group.WORLD
+  assert isinstance(world_pg, torch_xla.distributed.xla_backend.ProcessGroupXla)
+  world_pg.size = lambda: size
+
+
+def set_world_rank(rank):
+  world_pg = dist.group.WORLD
+  assert isinstance(world_pg, torch_xla.distributed.xla_backend.ProcessGroupXla)
+  world_pg.rank = lambda: rank
+
+
+def get_world_size():
+  return dist.group.WORLD.size()
+
+
+@contextmanager
+def new_group_barrier_disabled():
+  orig_barrier_fn = torch.distributed.distributed_c10d._store_based_barrier
+  torch.distributed.distributed_c10d._store_based_barrier = lambda x, y, z: None
+  try:
+    yield
+  finally:
+    torch.distributed.distributed_c10d._store_based_barrier = orig_barrier_fn
+
+
+@contextmanager
+def xm_cc_op_intercepted(cc_op):
+  orig_xm_cc_op = getattr(xm, cc_op)
+
+  def build_collective_context(groups):
+    cctx = xm.CollectiveContext()
+    # Here we need to manipulate cctx to force xm.all_reduce to generate
+    # the collective communication ops in HLO.
+    # We may need to .requires_interhost_reduce in future.
+    cctx.requires_intercore_reduce = True
+    cctx.intercore_group = groups
+    return cctx
+
+  def patched_cc_op(*args, **kwargs):
+    kwargs['cctx'] = build_collective_context(kwargs['groups'])
+    return orig_xm_cc_op(*args, **kwargs)
+
+  setattr(xm, cc_op, patched_cc_op)
+  try:
+    yield
+  finally:
+    setattr(xm, cc_op, orig_xm_cc_op)
+
+
+def test_xla_backend_exists():
+  # torch_xla.distributed._register_xla_backend() should have been
+  # automatically called.
+  pg_xla_creator = dist.Backend.XLA
+  assert pg_xla_creator is not None
+
+
+def test_process_group_creation():
+  pg_xla = get_process_group_xla(rank=1, size=2)
+  assert pg_xla is not None
+  assert pg_xla.rank() == 1
+  assert pg_xla.size() == 2
+
+
+def test_process_group_interface():
+  # Verifies ProcessGroupXla has all ProcessGroup public APIs.
+  # This test is needed because ProcessGroupXla cannot inherit from
+  # ProcessGroup (see ProcessGroupXla doc), thus we have to use a unit
```

Comment:
Can we state the reason that `ProcessGroupXla cannot inherit from ProcessGroup` here?

Reply:
Oh sorry, this was garbage that was needed before Shen's https://github.com/pytorch/pytorch/pull/66338. Now it's no longer needed. Will delete.